src/StarRating.tsx

```diff
@@ -103,33 +114,41 @@ const StarRating: React.FC<StarRatingProps> = ({
       } else {
         return getLeftColor(index);
       }
-    } else {
-      return getLeftColor(index);
     }
+
+    return getLeftColor(index);
   };
 
-  const getHoverRightColor = (index: number) => {
+  const getHoverRightColor = (index: number): DEFAULT_COLORS | string => {
     const isHalfSelectedHover = selectedRating % 1 === 0.5 && selectedStarIndex === index;
+
     if (isLeftSideHover) {
-      return isHalfSelectedHover ? DEFAULT_COLOR : index <= selectedStarIndex ? activeColor : DEFAULT_COLOR;
-    } else {
-      return isHalfSelectedHover ? activeColor : getLeftColor(index);
+      if (!isHalfSelectedHover && index > selectedRating) {
+        return activeColor;
+      }
+
+      return DEFAULT_COLORS.DEFAULT_COLOR;
     }
+
```

Comment:
```
if (isLeftSideHover) {
      if (!isHalfSelectedHover) {
        if (index > selectedRating)
          return activeColor;
        else
          return getLeftColor(index);
      }
      return DEFAULT_COLORS.DEFAULT_COLOR;
    }
```
I think we should return `getLeftColor(index)` in this condition `index <= selectedRating`.

Reply:
I think it's included.
Condition is just the opposite - `index > selectedRating` and in **else** `getLeftColor(index)` is returned.
But you can simply decline this PR 😉

Reply:
Sure, will double check that!
But will be available in 2 weeks if it's ok 😀